app/src/org/commcare/dalvik/activities/CommCareHomeActivity.java

```diff
@@ -549,9 +547,12 @@ else if(resultCode == RESULT_OK){
                 if (resultCode == RESULT_CANCELED) {
```

Comment:
@phillipm Talked to Clayton about this, he agreed with what you were saying that an exception should get thrown if the session was null, so it's most likely that the getNeededDatum() is in fact null. We discussed what's likely causing it, and strategy for at least handling the situation more carefully, so I'll grab this PR and implement that soon. If you're curious about what it is, it's a quick explanation, just long to type.